Potential build error with GCC 15

On my local system, building the module with GCC 15 fails:

```
[build] In file included from ./proxy/build/installed/install/include/proxy/proxy.ixx:3:
[build] ./proxy/build/installed/install/include/proxy/proxy.h:1527:16: error: ‘consteval auto pro::details::normalize(pro::proxiable_ptr_constraints)’ exposes TU-local entity ‘pro::details::invalid_size’
[build]  1527 | consteval auto normalize(proxiable_ptr_constraints value) {
[build]       |                ^~~~~~~~~
[build] ./proxy/build/installed/install/include/proxy/proxy.h:1524:23: note: ‘pro::details::invalid_size’ declared with internal linkage
[build]  1524 | constexpr std::size_t invalid_size = std::numeric_limits<std::size_t>::max();
[build]       |                       ^~~~~~~~~~~~
[build] ninja: build stopped: subcommand failed.
[proc] The command: /home/ubuntu/.local/bin/cmake --build ./build/vscode --config Debug --target all  -- exited with code: 1
[driver] Build completed: 00:00:01.560
[build] Build finished with exit code 1
```

```
> g++-15 --version
g++-15 (Ubuntu 15-20250404-0ubuntu1) 15.0.1 20250404 (experimental) [master r15-9193-g08e803aa9be]
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

> g++-14 --version
g++-14 (Ubuntu 14.2.0-19ubuntu2) 14.2.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

This issue doesn't show up with GCC 14 (the version CI is using). I'm investigating.

_Originally posted by @SidneyCogdill in https://github.com/microsoft/proxy/pull/293#discussion_r2090372381_

--- a/include/proxy/v4/proxy.h
+++ b/include/proxy/v4/proxy.h
@@ -1753,8 +1753,9 @@ private:
   T&& value_;
 };
 
-constexpr std::size_t invalid_size = std::numeric_limits<std::size_t>::max();
-constexpr constraint_level invalid_cl = static_cast<constraint_level>(
+inline constexpr std::size_t invalid_size =
+    std::numeric_limits<std::size_t>::max();
+inline constexpr constraint_level invalid_cl = static_cast<constraint_level>(
     std::numeric_limits<std::underlying_type_t<constraint_level>>::min());
 consteval std::size_t merge_size(std::size_t a, std::size_t b) {
   return a < b ? a : b;

--- a/tests/proxy_dispatch_tests.cpp
+++ b/tests/proxy_dispatch_tests.cpp
@@ -40,6 +40,11 @@ private:
   int value_;
 };
 
+struct TestFacadeBase : pro::facade_builder //
+                        ::add_convention<pro::operator_dispatch<"<<", true>,
+                                         std::ostream&(std::ostream& out)> //
+                        ::build {};
+
 PRO_DEF_FREE_AS_MEM_DISPATCH(FreeMemToString, std::to_string, ToString);
 
 } // namespace proxy_dispatch_tests_details
@@ -784,21 +789,17 @@ TEST(ProxyDispatchTests, TestIndirectConversion) {
 }
 
 TEST(ProxyDispatchTests, TestDirectConversion) {
-  struct TestFacadeBase : pro::facade_builder //
-                          ::add_convention<pro::operator_dispatch<"<<", true>,
-                                           std::ostream&(std::ostream & out)> //
-                          ::build {};
   struct TestFacade
       : pro::facade_builder                                           //
-        ::add_facade<TestFacadeBase>                                  //
+        ::add_facade<details::TestFacadeBase>                         //
         ::add_convention<pro::operator_dispatch<"+=">, void(int val)> //
         ::add_direct_convention<pro::conversion_dispatch,
-                                pro::proxy<TestFacadeBase>() &&> //
+                                pro::proxy<details::TestFacadeBase>() &&> //
         ::build {};
   pro::proxy<TestFacade> p1 = std::make_unique<int>(123);
   *p1 += 3;
-  pro::proxy<TestFacadeBase> p2 =
-      static_cast<pro::proxy<TestFacadeBase>>(std::move(p1));
+  pro::proxy<details::TestFacadeBase> p2 =
+      static_cast<pro::proxy<details::TestFacadeBase>>(std::move(p1));
   ASSERT_FALSE(p1.has_value());
   std::ostringstream stream;
   stream << *p2;

